### PR TITLE
Add support for WRF-ELEC

### PR DESCRIPTION
**Pull request by marcost2**
And also fix WRF@3.9.1.1 build

### CODE REVIEW

File: var/spack/repos/builtin/packages/wrf/package.py [hunk 1]
```diff
@@ -350,8 +390,13 @@ def run_compile_script(self):
...
 
-        # num of compile jobs capped at 20 in wrf
-        num_jobs = str(min(int(make_jobs), 10))
+        # WRF-ELEC as of version 3.9.1.1 WILL fail if compiled with more than
+        # 2 parallel instances
+        if self.spec.satisfies("+elec"):
+            num_jobs = str(2)
+        else:
+            # num of compile jobs capped at 20 in wrf
+            num_jobs = str(min(int(make_jobs), 10))
```
Comment (MichaelLaufer) [inline var/spack/repos/builtin/packages/wrf/package.py]: Is this a spack recipe issue or a WRF-ELEC one? Also, can you change the max num_jobs to 20 (it is set to 10 for no reason, while the comment shows 20).

File: var/spack/repos/builtin/packages/wrf/package.py [hunk 2]
```diff
@@ -111,13 +118,35 @@ class Wrf(Package):
...
+              msg="WRF_ELEC is only supported in V3.9.1.1")
+
+    conflicts("build_type=serial", when="+elec",
+              msg="WRF_ELEC only supports dmpar")
+
+    conflicts("build_type=smpar", when="+elec",
+              msg="WRF_ELEC only supports dmpar")
+
+    conflicts("build_type=dm+sm", when="+elec",
+              msg="WRF_ELEC only supports dmpar")
```
Comment (MichaelLaufer) [inline var/spack/repos/builtin/packages/wrf/package.py:138]: @tldahlgren is there a recommended way to merge these into a single conflict?

**Comment by MichaelLaufer** [comment]
The WRF package LGTM, aside from the two comments.

**Comment by haampie** [comment]
Please don't add the 25K line patch to Spack, can you host it elsewhere?

**Comment by marcost2** [comment]
> The WRF package LGTM, aside from the two comments.


What two comments?

> Please don't add the 25K line patch to Spack, can you host it elsewhere?

I've tried of thinking about an alternative way to achieve it. I COULD just use the downloaded resource and overwrite the files, but given that wrf-elec was built on an earlier version (v3.9). 
What would the better solution be here? That i host in my github a fork of wrf with everything already merged in and use that instead?

File: var/spack/repos/builtin/packages/boxmg4wrf/package.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    use with an upcoming release of WRF (Weather Research and Forecasting)
+    model including storm electrification. Only runs under MPI at this time
+    (i.e., no SMP/OpenMP support)."""
+
+    homepage = "https://sourceforge.net/projects/boxmg4wrf/"
+    git      = "https://git.code.sf.net/p/boxmg4wrf/git"
+
+    parallel = False
+
+    # maintainers = ['github_user1', 'github_user2']
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/boxmg4wrf/package.py]: This should be removed if you don't want to add yourself as a reviewer by including your GitHub account in the list.

**Review by tldahlgren — COMMENTED** [review]
We generally encourage upstreaming code changes.  That *is* a large patch.

File: var/spack/repos/builtin/packages/wrf/package.py (hunk 1 above)
Comment (marcost2) [reply var/spack/repos/builtin/packages/wrf/package.py]: It's a WRF-ELEC one; even when compiling standalone using anything higher than -j 2 will cause a cascade of errors.

Also done

File: var/spack/repos/builtin/packages/boxmg4wrf/package.py (hunk 3 above)
Comment (marcost2) [reply var/spack/repos/builtin/packages/boxmg4wrf/package.py]: Done

**Comment by marcost2** [comment]
> Please don't add the 25K line patch to Spack, can you host it elsewhere?

Would hosting it in a public facing server to be downloaded (AKA what Amber patches do) be a suitable compromise?

> We generally encourage upstreaming code changes. That _is_ a large patch.

That it is; however given the fact that wrf 3.9.1.1 is ancient (and this code isn't mine to begin with and is plagued with issues) it is not a possibility. The author of WRF-ELEC is working to merge it into the 4.3 codebase upstream so hopefully all of this will be obsolete soon

File: var/spack/repos/builtin/packages/wrf/package.py (hunk 2 above)
Comment (tldahlgren) [reply var/spack/repos/builtin/packages/wrf/package.py:138]: Unfortunately, I don't believe so.  You could put the directive in a for-loop over the build types to tighten the code.

**Comment by marcost2** [comment]
@haampie @tldahlgren 
Is this alternative (hosting the patches on a server) somewhat more acceptable?

**Comment by marcost2** [comment]
@tldahlgren @MichaelLaufer Sorry for bumping, but any updates on this?

**Comment by alalazo** [comment]
superseded by #42256

**Pull request closed without merging**